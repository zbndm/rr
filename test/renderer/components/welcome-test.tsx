import React from 'react';
import { Welcome } from '../../../src/renderer/components/welcome';
import renderer from 'react-test-renderer';

const mathBackup = global.Math;

describe('Welcome', () => {
  beforeAll(() => {
    const mockMath = Object.create(global.Math);
    mockMath.random = () => 0.5;
    global.Math = mockMath;
  });

  afterAll(() => {
    global.Math = mathBackup;
  });

  it('welcome renders correctly', () => {
    const tree = renderer.create(
      // tslint:disable-next-line:no-empty
      <Welcome sleuth=':sleuth:' state={{} as any} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
