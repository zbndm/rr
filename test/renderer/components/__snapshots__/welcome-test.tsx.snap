// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome welcome renders correctly 1`] = `
<div
  className="Welcome"
>
  <i />
  <div>
    <h1
      className="Title"
    >
      <span
        className="Emoji"
      >
        :sleuth:
      </span>
      <span>
        Sleuth
      </span>
    </h1>
    <h4>
      Drop a logs zip file or folder anywhere on this window to open it.
    </h4>
  </div>
  <div />
</div>
`;
